homeassistant/components/plugwise/gateway.py

```diff
@@ -163,49 +169,70 @@ async def async_unload_entry_gw(hass: HomeAssistant, entry: ConfigEntry):
     return unload_ok
 
 
+async def _update_listener(hass: HomeAssistant, entry: ConfigEntry):
+    """Handle options update."""
+    coordinator = hass.data[DOMAIN][entry.entry_id][COORDINATOR]
+    coordinator.update_interval = timedelta(
+        seconds=entry.options.get(CONF_SCAN_INTERVAL)
+    )
+
+
 class SmileGateway(CoordinatorEntity):
     """Represent Smile Gateway."""
 
-    def __init__(self, api, coordinator, name, dev_id):
+    def __init__(self, api, coordinator, dev_id, name, model, vendor, fw):
         """Initialise the gateway."""
         super().__init__(coordinator)
 
         self._api = api
-        self._name = name
+        self._coordinator = coordinator
         self._dev_id = dev_id
-
+        self._device_class = None
+        self._device_name = name
+        self._fw_version = fw
+        self._manufacturer = vendor
+        self._model = model
+        self._name = None
         self._unique_id = None
-        self._model = None
-
-        self._entity_name = self._name
 
     @property
-    def unique_id(self):
-        """Return a unique ID."""
-        return self._unique_id
+    def available(self):
+        """Return True if entity is available."""
+        return self.coordinator.last_update_success
 
     @property
-    def name(self):
-        """Return the name of the entity, if any."""
-        return self._name
+    def device_class(self):
+        """Return the class of this device, from component DEVICE_CLASSES."""
+        return self._device_class
 
     @property
-    def device_info(self) -> DeviceInfo:
+    def device_info(self) -> dict[str, any]:
         """Return the device information."""
         device_information = {
             "identifiers": {(DOMAIN, self._dev_id)},
-            "name": self._entity_name,
-            "manufacturer": "Plugwise",
+            "name": self._device_name,
+            "manufacturer": self._manufacturer,
+            "model": self._model,
+            "sw_version": self._fw_version,
         }
 
-        if self._model is not None:
-            device_information["model"] = self._model.replace("_", " ").title()
-
         if self._dev_id != self._api.gateway_id:
             device_information["via_device"] = (DOMAIN, self._api.gateway_id)
+        else:
+            device_information["name"] = f"Smile {self._api.smile_name}"
 
         return device_information
 
+    @property
+    def name(self):
+        """Return the name of the entity, if any."""
+        return self._name
+
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return self._unique_id
```

Comment:
`self._unique_id` doesn't seems to be used elsewhere, so you can use `_attr_unique_id` instead.

Reply:
Also in the works.